Controller/RoleController.php

```diff
@@ -116,8 +118,23 @@ public function createRoleAction(Request $request)
      */
     public function updateRoleAction(Request $request, $roleId)
     {
-        $role = $this->roleService->loadRole($roleId);
-        $roleData = (new RoleMapper())->mapToFormData($role);
+        try {
+            $roleDraft = $this->roleService->loadRoleDraft($roleId);
+        } catch (NotFoundException $e) {
+            // The draft doesn't exist, let's create one
+            try {
+                $role = $this->roleService->loadRole($roleId);
+                $roleDraft = $this->roleService->createRoleDraft($role);
+            } catch (UnauthorizedException $e) {
+                return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
+            } catch (NotFoundException $e) {
```

Comment:
You don't need to catch those any more (`NotFoundException` / `UnauthorizedException`) as it's now handled by Symfony under the hood. Besides, `eZPlatformUIBundle:Pjax:accessDenied` doesn't exist any more.